entity-filer/src/entity_filer/worker.py

```diff
@@ -57,7 +59,16 @@ def process_filing(payment_token, flask_app):
         raise QueueException('Flask App not available.')
 
     with flask_app.app_context():
-        filing_submission = get_filing_by_payment_id(payment_token['paymentToken'].get('id'))
+
+        # try to find the filing 5 times before putting back on the queue - in case payment token ends up on the queue
+        # before it is assigned to filing.
+        counter = 1
+        filing_submission = None
+        while not filing_submission and counter <= 5:
+            filing_submission = get_filing_by_payment_id(payment_token['paymentToken'].get('id'))
+            counter += 1
```

Comment:
This might not matter if get_filing_by_payment_id takes some time, but don't you need to wait a second before incrementing the counter? Otherwise won't it loop through super fast and then throw the exception?

Reply:
Yep you're right. I've added the check. From a few tests (with print statements) it gets to cycle 2 before it can process, so that's good, and better than running through all 5 iterations and putting it back on the queue.